sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/workflowentity/WorkflowEntity.java

```diff
@@ -356,6 +505,67 @@ public static WorkflowEntity.StepBuilder step(String name) {
     return new WorkflowEntity.StepBuilder(name);
   }
 
+  public static class StepConfig {
+    public final String stepName;
+    public final Optional<Duration> timeout;
+    public final Optional<RecoverStrategy<?>> recoverStrategy;
+
+    public StepConfig(String stepName, Optional<Duration> timeout, Optional<RecoverStrategy<?>> recoverStrategy) {
+      this.stepName = stepName;
+      this.timeout = timeout;
+      this.recoverStrategy = recoverStrategy;
+    }
+  }
+
+  public static class RecoverStrategy<T> {
+
+    public final int maxRetries;
+    public final String failoverToStepName;
+    public final Optional<T> failoverStepInput;
+
+    public RecoverStrategy(int maxRetries, String failoverToStepName, Optional<T> failoverStepInput) {
+      this.maxRetries = maxRetries;
+      this.failoverToStepName = failoverToStepName;
+      this.failoverStepInput = failoverStepInput;
+    }
+
+    /**
+     * Retry strategy without failover configuration
+     */
+    public static class MaxRetriesRecoverStrategy {
```

Comment:
Maybe we could call it simply `MaxRetries` since it's not sharing a common interface with `RecoveryStrategy`.